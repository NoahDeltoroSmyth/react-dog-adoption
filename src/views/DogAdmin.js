import React from 'react';
import DogForm from '../component/DogForm';
import { useDog } from '../Context/DogContext';
import { addDog } from '../services/dogRoute';

export default function DogAdmin() {
  const { name, breed, age, bio, image, setName, setBreed, setAge, setBio, setImage } = useDog();

  const handleSubmit = async (e) => {
    e.preventDefault();
    const { status } = await addDog(name, breed, age, bio, image);
    if ((name.length, breed.length, age.length, bio.length, image.length === 0)) {
      alert('You must add information about the doggo');
    } else if (status >= 400) {
      alert('Status error');
    } else {
      alert('Successfully added');
    }
    setName('');
    setBreed('');
    setAge('');
    setBio('');
    setImage('');
  };

  return (
    <div>
      <DogForm
        name={name}
        setName={setName}
        breed={breed}
        setBreed={setBreed}
        age={age}
        setAge={setAge}
        bio={bio}
        setBio={setBio}
        image={image}
        setImage={setImage}
        handleSubmit={handleSubmit}
      />
    </div>
  );
}
