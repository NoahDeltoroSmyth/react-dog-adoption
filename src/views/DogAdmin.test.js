import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import DogAdmin from './DogAdmin';

it('should render h1 text', async () => {
  //render
  const { container } = render(
    <MemoryRouter>
      <DogAdmin />
    </MemoryRouter>
  );
  //wait for load
  await screen.findByText('Dog Information');
  //take screen shot
  expect(container).toMatchSnapshot();
});
