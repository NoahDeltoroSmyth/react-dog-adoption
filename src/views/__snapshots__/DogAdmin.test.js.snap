// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render h1 text 1`] = `
<div>
  <div>
    <div
      class="form"
    >
      <form>
        <h1>
          Dog Information
        </h1>
        <div
          class="form-control"
        >
          <label>
            Name:
          </label>
          <input
            placeholder="Name"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Breed:
          </label>
          <input
            placeholder="Breed"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Age:
          </label>
          <input
            placeholder="Age"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Bio:
          </label>
          <input
            placeholder="Bio"
            type="text"
            value=""
          />
        </div>
        <div
          class="form-control"
        >
          <label>
            Image:
          </label>
          <input
            placeholder="Image"
            type="text"
            value=""
          />
        </div>
        <div>
          <button
            class="button"
          >
            Save
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
